rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java

```diff
@@ -1962,6 +1962,36 @@ void getPendingFederatorPublicKeyOfType_withSpecificRskAndMstKeys_returnsFederat
         }
     }
 
+    @Test
+    void getNewFederationBtcUTXOs_whenNoUTXOsWereSaved_returnsEmptyList() {
+        storageAccessor = new InMemoryStorage();
+        storageProvider = new FederationStorageProviderImpl(storageAccessor);
+        federationSupport = federationSupportBuilder
+            .withFederationConstants(federationMainnetConstants)
+            .withFederationStorageProvider(storageProvider)
+            .build();
+
+        List<UTXO> newFederationUTXOs = federationSupport.getNewFederationBtcUTXOs();
+        assertThat(newFederationUTXOs, is(Collections.emptyList()));
+    }
+
+    @Test
+    void getNewFederationBtcUTXOs_whenSavingUTXOs_returnsNewFederationUTXOs() {
+        storageAccessor = new InMemoryStorage();
+        storageProvider = new FederationStorageProviderImpl(storageAccessor);
+        federationSupport = federationSupportBuilder
+            .withFederationConstants(federationMainnetConstants)
+            .withFederationStorageProvider(storageProvider)
+            .build();
+        Address btcAddress = BitcoinTestUtils.createP2PKHAddress(federationMainnetConstants.getBtcParams(), "address");
```

Comment:
```suggestion
        Address btcFederationAddress = BitcoinTestUtils.createP2PKHAddress(federationMainnetConstants.getBtcParams(), "federation");
```

Reply:
i find that confusing, because that wont be the federation address